### PR TITLE
[+] Added tests for `BillUpdateUseCase`.

### PR DESCRIPTION
**Pull request by ScottCTD**
# Tasks

- [x] [+] Added tests for `BillUpdateUseCase`.
- [x] [+] Added and implemented some helper methods in DatabaseGateway. 
- [x] [*] Fixed many bugs.

### CODE REVIEW

File: src/main/java/billgates/database/MySQLDatabaseGateway.java [hunk 1]
```diff
@@ -471,6 +471,42 @@ INSERT INTO users (username, password, bill_id) VALUE (
         }
     }
 
+    @Override
+    public void deleteUser(QueryUserData user) {
```
Comment (Lei-Tin) [inline src/main/java/billgates/database/MySQLDatabaseGateway.java:475]: Do you not want to override a method for the ID and the Username? 

I think calling the method by passing an int or a username is easier than creating a `QueryUserData` class then throwing it in?

File: src/test/java/billgates/use_cases/BillUpdateUseCaseTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        this.gateway.insertUser(this.user.getQueryUserData());
+        this.gateway.createBillTable(this.user.getBillId());
+        this.presenter = new BillUpdatePresenter(
+                viewModel -> BillUpdateUseCaseTest.this.viewModel = viewModel);
+        this.useCase = new BillUpdateUseCase(this.presenter, this.gateway);
+        this.controller = new BillUpdateController(this.useCase);
+    }
+
+    @After
+    public void clean() {
```
Comment (Lei-Tin) [inline src/test/java/billgates/use_cases/BillUpdateUseCaseTest.java:52]: Do you want to call these `setUp` and `tearDown` by convention of JUnit Tests?

**Review by Lei-Tin — APPROVED** [review]
Looking good!

File: src/test/java/billgates/use_cases/BillUpdateUseCaseTest.java (hunk 2 above)
Comment (ScottCTD) [reply src/test/java/billgates/use_cases/BillUpdateUseCaseTest.java:52]: GOOD

File: src/main/java/billgates/database/MySQLDatabaseGateway.java (hunk 1 above)
Comment (ScottCTD) [reply src/main/java/billgates/database/MySQLDatabaseGateway.java:475]: Yes

**Review by Lei-Tin — APPROVED** [review]
GOOD
Also, good fix for the issues that I raised earlier today!

**Review by CharlotteChenyy — APPROVED** [review]
👍👍👍

**Pull request merged**